zebra-state/src/service/finalized_state/disk_format/transparent.rs

```diff
@@ -309,6 +329,94 @@ impl UnspentOutputAddressLocation {
     }
 }
 
+/// A single unspent output for a [`transparent::Address`].
+///
+/// We store both the address location key and unspend output location value
+/// in the RocksDB column family key. This improves insert and delete performance.
+///
+/// This requires 8 extra bytes for each unspent output,
+/// because we repeat the key for each value.
+/// But RocksDB compression reduces the duplicate data size on disk.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[cfg_attr(
+    any(test, feature = "proptest-impl"),
+    derive(Arbitrary, Serialize, Deserialize)
+)]
+pub struct AddressUnspentOutput {
+    /// The location of the first [`transparent::Output`] sent to the address in `output`.
+    address_location: AddressLocation,
+
+    /// The location of this unspent output.
+    unspent_output_location: OutputLocation,
+}
+
+impl AddressUnspentOutput {
+    /// Create a new [`AddressUnspentOutput`] from an address location,
+    /// and an unspent output location.
+    pub fn new(
+        address_location: AddressLocation,
+        unspent_output_location: OutputLocation,
+    ) -> AddressUnspentOutput {
+        AddressUnspentOutput {
+            address_location,
+            unspent_output_location,
+        }
+    }
+
+    /// Create an [`AddressUnspentOutput`] which starts iteration for the supplied address.
+    /// Used to look up the first output with [`ReadDisk::zs_next_key_value_from`].
+    ///
+    /// The unspent output location is before all unspent output locations in the index.
+    /// It is always invalid, due to the genesis consensus rules.
+    pub fn address_iterator_start(address_location: AddressLocation) -> AddressUnspentOutput {
+        // Iterating from the lowest possible output location gets us the first output.
+        let zero_output_location = OutputLocation::from_usize(Height(0), 0, 0);
+
+        AddressUnspentOutput {
+            address_location,
+            unspent_output_location: zero_output_location,
+        }
+    }
+
+    /// Update the unspent output location to the next possible output for the supplied address.
+    /// Used to look up the next output with [`ReadDisk::zs_next_key_value_from`].
+    ///
+    /// The updated unspent output location may be invalid.
```

Comment:
Suggestion to make it clearer:

```suggestion
    /// The updated unspent output location may be invalid, which is not an issue
    /// since [`ReadDisk::zs_next_key_value_from`] will fetch the next existing (valid) value.
```